src/sentry_envelope.c

```diff
@@ -234,6 +255,45 @@ sentry__envelope_add_event(sentry_envelope_t *envelope, sentry_value_t event)
     return item;
 }
 
+sentry_envelope_item_t *
+sentry__envelope_add_transaction(
+    sentry_envelope_t *envelope, sentry_value_t transaction)
+{
+    sentry_envelope_item_t *item = envelope_add_item(envelope);
+    if (!item) {
+        return NULL;
+    }
+
+    sentry_jsonwriter_t *jw = sentry__jsonwriter_new(NULL);
+    if (!jw) {
+        return NULL;
+    }
+
+    sentry_value_t event_id = sentry__ensure_event_id(transaction, NULL);
+
+    item->event = transaction;
+    sentry__jsonwriter_write_value(jw, transaction);
```

Comment:
this isn't really wrong because the value is still alive.   but rust wouldn't let you do this since `sentry__jsonwriter_write_value` borrows a value which you just passed ownership away from on the previous line.